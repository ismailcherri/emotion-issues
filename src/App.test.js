import { shallow } from "enzyme";
import styled from "@emotion/styled";

test("it does not break on self enclosed styled components", () => {
  const Styled = styled.div`
    background-color: grey;
  `;

  expect(
    shallow(
      <div>
        <Styled>Doesn't break here</Styled>
      </div>
    )
  ).toMatchSnapshot();
});

test("it breaks on self enclosed styled components", () => {
  const SelfEnclosed = styled.div`
    background-color: grey;
  `;

  expect(
    shallow(
      <div>
        <SelfEnclosed />
      </div>
    )
  ).toMatchSnapshot();
});

test("it breaks on HOC with children function call", () => {
  expect(shallow(<div>{() => <div>Some content</div>}</div>)).toMatchSnapshot();
});

test("it breaks on dive calls on styled component", () => {
  function Button({ children, ...props }) {
    return <button {...props}>{children}</button>;
  }

  const StyledButton = styled(Button)`
    background-color: grey;
  `;

  const wrapper = shallow(<StyledButton>Click me</StyledButton>);
  expect(wrapper.dive()).toMatchSnapshot();
});
