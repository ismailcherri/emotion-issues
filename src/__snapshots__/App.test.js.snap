// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it breaks on HOC with children function call 1`] = `
<div>
  <Component />
</div>
`;

exports[`it breaks on dive calls on styled component 1`] = `null`;

exports[`it breaks on dive calls on styled component 2`] = `<Bar />`;

exports[`it breaks on dive calls on styled component 3`] = `null`;

exports[`it breaks on self enclosed styled components 1`] = `
<div>
  <Styled(div) />
</div>
`;

exports[`it does not break on self enclosed styled components 1`] = `
<div>
  <Styled(div)>
    Doesn't break here
  </Styled(div)>
</div>
`;
